lib/Dialect/FIRRTL/Transforms/Dedup.cpp

```diff
@@ -939,6 +939,12 @@ class DedupPass : public DedupBase<DedupPass> {
   void runOnOperation() override {
     auto *context = &getContext();
     auto circuit = getOperation();
+
+    // If this circuit is marked "NoDedup" skip this pass.
+    if (AnnotationSet::removeAnnotations(
+            circuit, "firrtl.transforms.NoCircuitDedupAnnotation"))
+      return markAllAnalysesPreserved();
```

Comment:
It probably makes sense to remove this given the pipeline we have. One thing to consider, should this also remove any `NoDedupAnnotation`s that may be placed on modules in the circuit, even if deduplication is skipped? I can see a case for and against and am fine with whatever you decide.